Cabal/Distribution/Simple/Register.hs

```diff
@@ -441,7 +463,11 @@ generalInstalledPackageInfo adjustRelIncDirs pkg abi_hash lib lbi clbi installDi
     hasModules = not $ null (allLibModules lib clbi)
     hasLibrary = hasModules || not (null (cSources bi))
                             || (not (null (jsSources bi)) &&
-                                compilerFlavor (compiler lbi) == GHCJS)
+                                compilerFlavor comp == GHCJS)
+    comp = compiler lbi
+    supportsLibraryDynDir = case compilerFlavor comp of
+      GHC -> compilerVersion comp > mkVersion [8,0,1]
+      _   -> False
```

Comment:
I created a function `D.S.Compiler.supportsLibraryDynDir` in the latest version of the PR.